src/hotspot/cpu/x86/gc/shared/barrierSetNMethod_x86.cpp

```diff
@@ -170,9 +180,18 @@ static const int entry_barrier_offset(nmethod* nm) {
 }
 
 static NativeNMethodCmpBarrier* native_nmethod_barrier(nmethod* nm) {
-  address barrier_address = nm->code_begin() + nm->frame_complete_offset() + entry_barrier_offset(nm);
+  address barrier_address;
+#if INCLUDE_JVMCI
+  if (nm->is_compiled_by_jvmci()) {
+    barrier_address = nm->code_begin() + nm->jvmci_nmethod_data()->nmethod_entry_patch_offset();
+  } else
+#endif
+    {
+      barrier_address = nm->code_begin() + nm->frame_complete_offset() + entry_barrier_offset(nm);
+    }
+
   NativeNMethodCmpBarrier* barrier = reinterpret_cast<NativeNMethodCmpBarrier*>(barrier_address);
-  debug_only(barrier->verify());
+  barrier->verify();
```

Comment:
I think this should be reverted to:
`debug_only(barrier->verify());`

Reply:
verify now contains only an assert so the debug_only is unnecessary

Reply:
I see. Still, `verify()` also contains `err_msg("%s", "");` and that calls `jio_vsnprintf`:
```c++
template <size_t bufsz>
FormatBuffer<bufsz>::FormatBuffer(const char * format, ...) : FormatBufferBase(_buffer) {
  va_list argp;
  va_start(argp, format);
  jio_vsnprintf(_buf, bufsz, format, argp);
  va_end(argp);
}
```
So I think we should probably guard `err_msg("%s", "")` as well if we want to ensure there's no overhead in a product build.